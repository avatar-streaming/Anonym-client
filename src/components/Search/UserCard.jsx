import React from "react";
import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

function UserCard({ _id, thumnail, userName, followers }) {
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
