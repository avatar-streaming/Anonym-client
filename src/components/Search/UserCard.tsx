import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

<<<<<<< HEAD:src/components/Search/UserCard.tsx
interface UserCardProps {
  _id: string;
  thumnail: string;
  userName: string;
  followers: string[];
}

function UserCard({ _id, thumnail, userName, followers }: UserCardProps) {
=======
function UserCard({ _id, thumnail, userName, followers }) {
>>>>>>> c05b8641dad7cfa992c68cc1b3c69d422f4bf8ce:src/components/Search/UserCard.jsx
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
