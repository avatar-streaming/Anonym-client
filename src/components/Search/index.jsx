import React from "react";
import UserCard from "./UserCard";
import useSearchUsers from "../../hooks/useSearchUsers";

function Search() {
  const userList = useSearchUsers();

  return (
    <div className="content-wrapper">
      <div>
        {userList.map((user) => (
          <UserCard key={user._id} user={user} />
        ))}
      </div>
    </div>
  );
}

export default Search;
