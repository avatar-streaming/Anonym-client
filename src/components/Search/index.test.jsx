import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { Router } from "react-router";
import Search from ".";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";

describe("<Search />", () => {
  const mockStore = configureMockStore([thunk]);
  const history = createMemoryHistory();
  let store = null;
  let component = null;

  beforeEach(() => {
    store = mockStore({
      user: {
        userInfo: {
          "_id": 0,
          userName: "user 0",
          thumnail: "thumnail 0",
          followers: [],
        },
      },
      search: {
        searchList: [
          {
            "_id": 1,
            userName: "user 1",
            thumnail: "thumnail 1",
            followers: [],
          },
          {
            "_id": 2,
            userName: "user 2",
            thumnail: "thumnail 2",
            followers: [],
          },
        ],
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Search />
        </Router>
      </Provider>
    );
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
