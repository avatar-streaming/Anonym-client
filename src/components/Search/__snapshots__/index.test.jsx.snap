// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render with given state from redux store 1`] = `
<div
  className="content-wrapper"
>
  <div>
    <div
      className="user-card"
    >
      <div
        className="user-card__left"
      >
        <img
          alt="user thumnail"
          className="user-card__thumnail user-thumnail"
          src="thumnail 1"
        />
      </div>
      <div
        className="user-card__right"
      >
        <div
          className="user-card__description"
        >
          <div
            className="user-card__user-name"
          >
            user 1
          </div>
          <div
            className="follower-number"
          >
            0
             followers
          </div>
        </div>
        <div
          className="follow-button-box"
        >
          <button
            className="follow-button"
            onClick={[Function]}
          >
            follow
          </button>
        </div>
      </div>
    </div>
    <div
      className="user-card"
    >
      <div
        className="user-card__left"
      >
        <img
          alt="user thumnail"
          className="user-card__thumnail user-thumnail"
          src="thumnail 2"
        />
      </div>
      <div
        className="user-card__right"
      >
        <div
          className="user-card__description"
        >
          <div
            className="user-card__user-name"
          >
            user 2
          </div>
          <div
            className="follower-number"
          >
            0
             followers
          </div>
        </div>
        <div
          className="follow-button-box"
        >
          <button
            className="follow-button"
            onClick={[Function]}
          >
            follow
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
